### PR TITLE
[enhancement](*): add checkstyle file

### PR DESCRIPTION
**Pull request by jackwener**
# Proposed changes

Issue Number: related #8985 

## Problem Summary:

add checkstyle file

## Checklist(Required)

1. Does it affect the original behavior: (Yes)
2. Has unit tests been added: (No Need)
3. Has document been added or modified: (No Need)
4. Does it need to update dependencies: (No)
5. Are there any changes that cannot be rolled back: (No)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by jackwener** [comment]
It's a base file. We can add more rule for it.

**Comment by jackwener** [comment]
PTAL @EmmyMiao87 @morrySnow @wangshuo128

**Comment by cody195** [comment]
Hi @jackwener, it is not a good idea to add `checkstyle` file with so many rules directly. The project will always build failed.

image

**Comment by jackwener** [comment]
Yes, it,s just for review.

Charles ***@***.***> 于 2022年4月19日周二 09:05写道：

File: fe/checkstyle.xml [hunk 1]
```diff
@@ -26,8 +26,173 @@
...
+            <message key="import.redundancy"
+                     value="Redundant import {0}."/>
+        </module>
+        <module name="UnusedImports">
+            <property name="severity" value="error"/>
+            <message key="import.unused"
+                     value="Unused import: {0}."/>
+        </module>
+        <module name="ImportOrder">
+            <property name="groups" value="org.apache.doris,org.apache,*,javax,java"/>
```
Comment (EmmyMiao87) [inline fe/checkstyle.xml:171]: The import order should be like this 
```
4=javax
3=java
2=org
1=com
0=org.apache.doris
```

File: fe/checkstyle.xml [hunk 2]
```diff
@@ -26,8 +26,196 @@
...
+            <message key="import.redundancy"
+                     value="Redundant import {0}."/>
+        </module>
+        <module name="UnusedImports">
+            <property name="severity" value="error"/>
+            <message key="import.unused"
+                     value="Unused import: {0}."/>
+        </module>
+        <module name="ImportOrder">
+            <property name="groups" value="org.apache.doris,com,org,java,*"/>
```
Comment (EmmyMiao87) [inline fe/checkstyle.xml:157]:
```suggestion
            <property name="groups" value="org.apache.doris,com.google.common,org,java,*"/>
```

**Review by EmmyMiao87 — APPROVED** [review]
LGTM

**Comment by cody195** [comment]
> LGTM

Hi @EmmyMiao87, it is not a good idea to add `checkstyle` file with so many rules directly. The project already build failed, why you approved these changes?

image

**Comment by jackwener** [comment]
@cody195 Your consideration is right.

I made a draft #9113, I found there are some problem need to consider.

We should continue pushing this job in #8985. 

If you'd like take part in it, you can care it in #8985

**Pull request closed without merging**